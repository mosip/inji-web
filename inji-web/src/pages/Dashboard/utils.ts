export const convertStringIntoPascalCase = (text: string | null) => {
    if (text != null) {
        return text
            .toLocaleLowerCase()
            .split(" ")
            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
            .join(" ");
    }
};
