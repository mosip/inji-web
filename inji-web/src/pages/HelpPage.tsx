import React from "react";
import {HelpAccordion} from "../components/Help/HelpAccordion";
import {NavBar} from "../components/Common/NavBar";
import {useTranslation} from "react-i18next";

export const HelpPage: React.FC = () => {
    const {t} = useTranslation("HelpPage")
    return <React.Fragment>
        <div className={"bg-iw-background"} data-testid="Help-Page-Container">
            <NavBar title={t("title")} search={false} link={"/"}/>
            <div className="container mx-auto mt-8 ">
                <HelpAccordion/>
            </div>
        </div>
    </React.Fragment>
}
