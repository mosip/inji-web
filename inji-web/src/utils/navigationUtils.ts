import {ROUTES} from "../constants/Routes";

export const navigateToUserHome = (navigate: any) =>
    navigate(ROUTES.USER_HOME);