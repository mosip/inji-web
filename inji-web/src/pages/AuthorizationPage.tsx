import React, {useEffect, useState} from "react";
import {SpinningLoader} from "../components/Common/SpinningLoader";
import {api} from "../utils/api";
import {LandingPageWrapper} from "../components/Common/LandingPageWrapper";
import {ErrorSheildIcon} from "../components/Common/ErrorSheildIcon";
import {useTranslation} from "react-i18next";

export const AuthorizationPage: React.FC = () => {
    const url = window.location.href;
    const currentQueryParams = new URLSearchParams(window.location.search);
    const [error, setError] = useState<String>(currentQueryParams.get("error")+"");
    const {t} = useTranslation("AuthorizationPage");

    useEffect( () => {
        if(url.indexOf("error") === -1) {
            window.location.href = api.mimotoHost + "/authorize" + window.location.search;
        }
    },[])

    if(url.indexOf("error") == -1){
        return <div><LandingPageWrapper icon={<SpinningLoader/>} title={""} subTitle={""} gotoHome={false}/></div>
    }
    return <div><LandingPageWrapper icon={<ErrorSheildIcon />} title={t(`error.code.${error}`)} subTitle={t(`error.message.${error}`)} gotoHome={true}/></div>
}


