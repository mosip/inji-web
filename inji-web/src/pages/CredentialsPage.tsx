import React, {useEffect, useState} from "react";
import {useParams} from "react-router-dom";
import {RequestStatus, useFetch} from "../hooks/useFetch";
import {NavBar} from "../components/Common/NavBar";
import {CredentialList} from "../components/Credentials/CredentialList";
import {useDispatch, useSelector} from "react-redux";
import {storeSelectedIssuer} from "../redux/reducers/issuersReducer";
import {storeCredentials, storeFilteredCredentials} from "../redux/reducers/credentialsReducer";
import {api} from "../utils/api";
import {useTranslation} from "react-i18next";
import {toast} from "react-toastify";

import {ApiRequest, DisplayArrayObject, IssuerObject} from "../types/data";
import {getObjectForCurrentLanguage} from "../utils/i18n";
import {RootState} from "../types/redux";
import {isObjectEmpty} from "../utils/misc";

export const CredentialsPage: React.FC = () => {

    const {state, fetchRequest} = useFetch();
    const params = useParams<CredentialParamProps>();
    const dispatch = useDispatch();
    const {t} = useTranslation("CredentialsPage");
    const language = useSelector((state: RootState) => state.common.language);
    let displayObject= {} as DisplayArrayObject;
    let [selectedIssuer, setSelectedIssuer] = useState({} as IssuerObject)
    if(!isObjectEmpty(selectedIssuer)){
        displayObject = getObjectForCurrentLanguage(selectedIssuer.display, language);
    }

    useEffect(() => {
        const fetchCall = async () => {
            let apiRequest: ApiRequest = api.fetchSpecificIssuer;
            let response = await fetchRequest(
                apiRequest.url(params.issuerId ?? ""),
                apiRequest.methodType,
                apiRequest.headers()
            );
            dispatch(storeSelectedIssuer(response?.response));
            setSelectedIssuer(response?.response);

            apiRequest = api.fetchCredentialTypes2;
            response = await fetchRequest(
                apiRequest.url(params.issuerId ?? ""),
                apiRequest.methodType,
                apiRequest.headers()
            );
            // apiRequest = api.fetchCredentialTypes;
            // response = await fetchRequest(
            //     apiRequest.url(response?.response[".well-known"]),
            //     apiRequest.methodType,
            //     apiRequest.headers()
            // );
            dispatch(storeFilteredCredentials(response?.response?.supportedCredentials));
            dispatch(storeCredentials(response?.response?.supportedCredentials));
        }
        fetchCall();
    }, [])

    if (state === RequestStatus.ERROR) {
        toast.error(t("errorContent"));
    }


    return <React.Fragment>
        <div className="bg-iw-background min-h-screen"
             data-testid="Credentials-Page-Container">
            <NavBar title={displayObject?.name} search={true} fetchRequest={fetchRequest} link={"/"}/>
            <div data-testid="Credential-List-Container" className="container mx-auto mt-8 px-10 sm:px-0">
                <CredentialList state={state}/>
            </div>
        </div>
    </React.Fragment>
}

