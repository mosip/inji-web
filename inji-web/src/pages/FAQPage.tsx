import React from 'react';
import {FAQAccordion} from '../components/Faq/FAQAccordion';
import {useTranslation} from 'react-i18next';
import {useLocation, useNavigate} from 'react-router-dom';
import {FAQPageProps} from '../components/Dashboard/types';
import {navigateToDashboardHome} from './Dashboard/utils';
import {PageTitle} from "../components/Common/PageTitle/PageTitle";
import { TertiaryButton } from './Dashboard/TertiaryButton';

export const FAQPage: React.FC<FAQPageProps> = ({backUrl, withHome}) => {
    const {t} = useTranslation(['FAQPage', 'Dashboard']);
    const navigate = useNavigate();
    const location = useLocation();
    const previousPagePath = location.state?.from;

    const handleBackClick = () => {
        if (backUrl) {
            navigate(backUrl); // Navigate to the URL sent by the parent
        } else if (previousPagePath) {
            navigate(previousPagePath); // Navigate to the previous link in history
        } else {
            navigateToDashboardHome(navigate); // Navigate to homepage if opened directly
        }
    };

    return (
        <div className="w-full flex flex-col sm:flex-row justify-between items-start p-1 sm:p-2 md:p-4 flex-grow overflow-y-auto">
            <div className="flex flex-row items-start gap-4 w-full">
                <div className="flex items-start">
                    <svg
                        width="29"
                        height="29"
                        viewBox="0 0 24 18"
                        fill="none"
                        xmlns="http://www.w3.org/2000/svg"
                        className="mr-2 cursor-pointer"
                        onClick={handleBackClick}
                    >
                        <path
                            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                            fill="#000000"
                        />
                    </svg>
                </div>
                <div className="flex flex-col items-start gap-8 w-full mr-2 sm:mr-4 md:mr-8 lg:mr-10">
                    <div className="flex flex-col items-start">
                        <PageTitle value={t('title')} testId={"faq"} />
                        {withHome && (
                            <TertiaryButton testId={'FAQ-Home-Button'} onClick={()=>navigateToDashboardHome(navigate)} title={t('Dashboard:Home.title')}/>
                        )}
                    </div>
                    <div className="w-full">
                        <FAQAccordion />
                    </div>
                </div>
            </div>
        </div>
    );
};
