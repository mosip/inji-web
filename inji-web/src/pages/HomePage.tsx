import React, {useEffect} from "react";
import {RequestStatus, useFetch} from "../hooks/useFetch";
import {IntroBox} from "../components/Common/IntroBox";
import {SearchIssuer} from "../components/Issuers/SearchIssuer";
import {IssuersList} from "../components/Issuers/IssuersList";
import {useDispatch} from "react-redux";
import {storeIssuers} from "../redux/reducers/issuersReducer";
import {api, MethodType} from "../utils/api";
import {IssuerObject} from "../types/data";
import {HeaderTile} from "../components/Common/HeaderTile";
import {useTranslation} from "react-i18next";
import {toast} from "react-toastify";

export const HomePage: React.FC = () => {

    const {state, fetchRequest} = useFetch();
    const dispatch = useDispatch();
    const {t} = useTranslation("HomePage");

    useEffect(() => {
        const fetchCall = async () => {
            const response = await fetchRequest(api.fetchIssuers(), MethodType.GET, null);
            const issuers = response?.response?.issuers.filter((issuer: IssuerObject) => issuer.credential_issuer === "Sunbird")
            dispatch(storeIssuers(issuers));
        }
        fetchCall();
    }, [])

    if (state === RequestStatus.ERROR) {
        toast.error(t("errorContent"));
    }

    return <div data-testid="Home-Page-Container">
        <div className="container mx-auto mt-8 px-4 flex-1 flex flex-col">
            <IntroBox/>
            <SearchIssuer state={state} fetchRequest={fetchRequest}/>
        </div>
        <HeaderTile content={t("containerHeading")}/>
        <IssuersList state={state}/>
    </div>

}
