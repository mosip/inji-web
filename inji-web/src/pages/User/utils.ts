import { User } from "../../components/Dashboard/types";
import { KEYS } from "../../utils/constants";
import {ROUTES} from "../../constants/Routes";

export const convertStringIntoPascalCase = (text: string | undefined) => {
    return (
        text?.toLocaleLowerCase()
            .split(' ')
            .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
            .join(' ')
    );
};

export const navigateToUserHome = (navigate: any) =>
    navigate(ROUTES.USER_HOME);

export const validateWalletUnlockStatus = (
    cachedWalletId: string | null,
    storageWalletId: string | null,
    navigate: (path: string) => void,
    user: User
) => {
    if (cachedWalletId && (cachedWalletId === storageWalletId)) {
        console.info('Wallet is unlocked!');
    } else {
        console.warn(
            'Wallet exists but is locked, redirecting to `/pin` to unlock the wallet.'
        );
        if(user) {
            navigate('/pin');
            localStorage.removeItem(KEYS.WALLET_ID);
        }
    }
};