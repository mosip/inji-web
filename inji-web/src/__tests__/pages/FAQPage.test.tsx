import { render, screen, fireEvent } from '@testing-library/react';
import { FAQPage } from '../../pages/FAQPage';
import { useLocation } from 'react-router-dom';
import { navigateToUserHome } from '../../utils/navigationUtils';
// Mocks
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => key,
  }),
}));

const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useLocation: jest.fn(),
}));

jest.mock('../../utils/navigationUtils', () => ({
  navigateToUserHome: jest.fn(),
}));
// is it correct?
jest.mock('../../components/Common/PageTitle/PageTitle', () => ({
  PageTitle: ({ value, testId }: { value: string; testId?: string }) => (
    <h1 data-testid={testId}>{value}</h1>
  ),
}));

jest.mock('../../components/Common/Buttons/TertiaryButton', () => ({
  TertiaryButton: ({ testId, onClick, title }: any) => (
    <button data-testid={testId} onClick={onClick}>{title}</button>
  ),
}));

jest.mock('../../components/Faq/FAQAccordion', () => ({
  FAQAccordion: () => <div data-testid="faq-accordion" />,
}));

describe('FAQPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders PageTitle and FAQAccordion', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: {} });

    render(<FAQPage backUrl={undefined} withHome={false} />);

    expect(screen.getByTestId('faq')).toHaveTextContent('title');
    expect(screen.getByTestId('faq-accordion')).toBeInTheDocument();
    expect(screen.queryByTestId('faq-home-button')).not.toBeInTheDocument();
  });

  it('renders home button if withHome=true and calls navigateToDashboardHome on click', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: {} });

    render(<FAQPage backUrl={undefined} withHome={true} />);
    const homeButton = screen.getByTestId('faq-home-button');
    expect(homeButton).toBeInTheDocument();
    expect(homeButton).toHaveTextContent('User:Home.title');

    fireEvent.click(homeButton);
    expect(navigateToUserHome).toHaveBeenCalledWith(mockNavigate);
  });

  it('clicking back arrow navigates to backUrl if present', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: {} });
  
    render(<FAQPage backUrl="/custom-back" withHome={false} />);
    const svg = document.querySelector('svg');
    expect(svg).toBeInTheDocument();
  
    fireEvent.click(svg!);
    expect(mockNavigate).toHaveBeenCalledWith('/custom-back');
  });

  it('clicking back arrow navigates to previousPagePath from location.state if no backUrl', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: { from: '/prev-page' } });

    render(<FAQPage backUrl={undefined} withHome={false} />);
    const svg = document.querySelector('svg');
    fireEvent.click(svg!);

    expect(mockNavigate).toHaveBeenCalledWith('/prev-page');
  });

  it('clicking back arrow calls navigateToDashboardHome if no backUrl or previousPagePath', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: {} });

    render(<FAQPage backUrl={undefined} withHome={false} />);
    const svg = document.querySelector('svg');
    fireEvent.click(svg!);

    expect(navigateToUserHome).toHaveBeenCalledWith(mockNavigate);
  });

  it('matches snapshot when withHome=false', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: {} });

    const { asFragment } = render(<FAQPage backUrl={undefined} withHome={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when withHome=true', () => {
    (useLocation as jest.Mock).mockReturnValue({ state: {} });

    const { asFragment } = render(<FAQPage backUrl={undefined} withHome={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});