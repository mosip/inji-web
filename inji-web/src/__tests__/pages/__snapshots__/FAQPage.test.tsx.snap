// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQPage matches snapshot when withHome=false 1`] = `
<DocumentFragment>
  <div
    class="w-full flex flex-col sm:flex-row justify-between items-start p-1 sm:p-2 md:p-4 flex-grow overflow-y-auto"
  >
    <div
      class="flex flex-row items-start gap-4 w-full"
    >
      <div
        class="flex items-start"
      >
        <svg
          class="mr-2 cursor-pointer"
          fill="none"
          height="29"
          viewBox="0 0 24 18"
          width="29"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            fill="#000000"
          />
        </svg>
      </div>
      <div
        class="flex flex-col items-start gap-8 w-full mr-2 sm:mr-4 md:mr-8 lg:mr-10"
      >
        <div
          class="flex flex-col items-start"
        >
          <h1
            data-testid="faq"
          >
            title
          </h1>
        </div>
        <div
          class="w-full"
        >
          <div
            data-testid="faq-accordion"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FAQPage matches snapshot when withHome=true 1`] = `
<DocumentFragment>
  <div
    class="w-full flex flex-col sm:flex-row justify-between items-start p-1 sm:p-2 md:p-4 flex-grow overflow-y-auto"
  >
    <div
      class="flex flex-row items-start gap-4 w-full"
    >
      <div
        class="flex items-start"
      >
        <svg
          class="mr-2 cursor-pointer"
          fill="none"
          height="29"
          viewBox="0 0 24 18"
          width="29"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            fill="#000000"
          />
        </svg>
      </div>
      <div
        class="flex flex-col items-start gap-8 w-full mr-2 sm:mr-4 md:mr-8 lg:mr-10"
      >
        <div
          class="flex flex-col items-start"
        >
          <h1
            data-testid="faq"
          >
            title
          </h1>
          <button
            data-testid="faq-home-button"
          >
            User:Home.title
          </button>
        </div>
        <div
          class="w-full"
        >
          <div
            data-testid="faq-accordion"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
